src/encoded/static/components/vis_defines.js

```diff
@@ -191,3 +202,260 @@ const browserOrder = [
+ * URLs.
+ * @param {object} results Search results
+ * @param {string} hostName Domain name of host
+ *
+ * @return {string} hub URL used in visualization URLs
+ */
+const generateBatchHubUrl = (results, hostName) => {
+    const parsedUrl = url.parse(results['@id']).search;
+    if (parsedUrl) {
+        return `${hostName}/batch_hub/${encodeURIComponent(parsedUrl.substr(1).replace('&', ',,'))}/hub.txt`;
+    }
+    return null;
+};
+
+
+/**
+ * Generate a list of relevant assemblies from the given search results. This includes any
+ * assemblies included in the search results, filtered by any specified in the query string that
+ * generated these search results, or all included assemblies if no assemblies were given in the
+ * query string.
+ * @param {object} results Search results
+ *
+ * @return {array} Assemblies in search results, filtered by query string assemblies if any
+ */
+const getRelevantAssemblies = (results) => {
+    let relevantAssemblies = [];
+    const assemblyFacet = results.facets.find(facet => facet.field === 'assembly');
+    if (assemblyFacet) {
+        // Get array of assemblies specified in the search query; empty array if none.
+        const specifiedAssemblies = results.filters.filter(filter => filter.field === 'assembly').map(filter => filter.term);
+        relevantAssemblies = assemblyFacet.terms.filter(term => term.doc_count > 0 && (specifiedAssemblies.length === 0 || specifiedAssemblies.includes(term.key)));
+    }
+    return relevantAssemblies.map(assembly => assembly.key).sort((a, b) => globals.assemblyPriority.indexOf(a) - globals.assemblyPriority.indexOf(b));
+};
+
+
+/**
+ * Currently this maps an assembly as stored in encode to an assembly that UCSC uses in the URL.
+ * Each mapping results in an object to allow for future expansion.
+ */
+const ucscAssemblyDetails = {
+    GRCh38: { mappedAssembly: 'hg38' },
+    'GRCh38-minimal': { mappedAssembly: 'hg38' },
+    hg19: { mappedAssembly: 'hg19' },
+    mm10: { mappedAssembly: 'mm10' },
+    'mm10-minimal': { mappedAssembly: 'mm10' },
+    mm9: { mappedAssembly: 'mm9' },
+    dm6: { mappedAssembly: 'dm6' },
+    dm3: { mappedAssembly: 'dm3' },
+    ce11: { mappedAssembly: 'ce11' },
+    ce10: { mappedAssembly: 'ce10' },
+    ce6: { mappedAssembly: 'ce6' },
+};
+
+
+/**
+ * Generate a batch hub URL for UCSC based on the given assembly.
+ * @param {string} assembly Assembly to use for the generated URL
+ * @param {object} {
+ *      visualizer {string} "UCSC" visualizer name to use in UI
+ *      url {string} URL to pass to UCSC to visualize the given assembly
+ * }
+ */
+const ucscUrlGenerator = (assembly, batchHubUrl) => {
+    const details = ucscAssemblyDetails[assembly];
+    if (details) {
+        return {
+            visualizer: 'UCSC',
+            url: `http://genome.ucsc.edu/cgi-bin/hgTracks?hubClear=${batchHubUrl}&db=${details.mappedAssembly}`,
+        };
+    }
+    return null;
```

Comment:
(Minor) Is this needed in that functions return undefined by default.

Reply:
I’m not a big fan of returning nothing from a function that’s supposed to return something, but I have no issues with those who do that.